SABT_MainUnit/MD13.c

```diff
@@ -0,0 +1,283 @@
+            break;
+            
+        case STATE_PROMPT:
+            md13_play_question();
+            md_next_state = STATE_INPUT;
+            break;
+            
+        case STATE_INPUT:
+            if (io_user_abort == true) {
+                PRINTF("[MD13] User aborted input\n\r");
+                md_next_state = STATE_REPROMPT;
+                io_init();
+                break;
+            }
+            if (get_number(&md_input_valid, &md_usr_res)) {
+                if (md_input_valid) {
+                    sprintf(dbgstr, "[MD13] User answer: %d\n\r", md_usr_res);
+                    PRINTF(dbgstr);
+                    play_mp3(MODE_FILESET, MP3_YOU_ANSWERED);
+                    play_number(md_usr_res);
+                    md_next_state = STATE_CHECKANS;
+                } else {
+                    PRINTF("[MD13] IO error\n\r");
+                }
+            }
+            break;
+            
+        case STATE_CHECKANS:
+            if ((md_usr_res >= md_op_1 && md_usr_res <= md_op_2) ||
+                 (md_usr_res >= md_op_2 && md_usr_res <= md_op_1)) {
+                // Correct answer
+                md_incorrect_tries = 0;
+                play_mp3(LANG_FILESET, MP3_CORRECT);
+                play_mp3(SYS_FILESET, MP3_TADA);
+                md_next_state = STATE_LVLSEL;
+            } else {
+                // Wrong answer
+                md_incorrect_tries++;
+                play_mp3(LANG_FILESET, MP3_INCORRECT);
+                if (md_incorrect_tries >= MAX_INCORRECT_TRIES) {
+                    //md13_play_answer();
+                }
+                play_mp3(LANG_FILESET, MP3_TRY_AGAIN);
+                md_next_state = STATE_PROMPT;
+            }
+            break;
+            
+        case STATE_REPROMPT:
+            md_last_dot = create_dialog(MP3_SKIP,
+                                        ENTER_CANCEL | LEFT_RIGHT);
+            switch (md_last_dot) {
+                    
+                case NO_DOTS:
+                    break;
+                    
+                    // Playing answer
+                case RIGHT: case LEFT:
+                    //md9_play_answer();
+                    break;
+                    
+                    // Skipping question
+                case ENTER:
+                    md_next_state = STATE_GENQUES;
+                    break;
+                    
+                    // Try again
+                case CANCEL:
+                    md_next_state = STATE_PROMPT;
+                    break;
+                    
+                default:
+                    break;
+            }
+            break;
+            
+        default:
+            PRINTF("[MD13] Error: next_state: ");
+            SENDBYTE(md_next_state);
+            NEWLINE;
+            quit_mode();
+            break;
+    }
+}
+
+void md13_call_mode_yes_answer(void) {
+    
+}
+
+void md13_call_mode_no_answer(void) {
+    
+}
+
+void md13_call_mode_left(void) {
+    
```

Comment:
I guess the instruction audio is confusing, left and right buttons can only be used in the reprompt state. I'll work on making them an interrupt event after I finished my work on SD card. I changed the order of audio now, is it clearer?